src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int anInt = random.nextInt(5);
+        return switch (anInt) {
+            case 1 -> new Circle(Colors.WHITE.name(), 10);
+            case 2 -> new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                    random.nextInt(10), random.nextInt(10));
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. For example, `int base = random.nextInt(10); int height = random.nextInt(10);` then pass `base` and `height` to the constructor of `IsoscelesTrapezoid`.